### PR TITLE
Implement :defined CSS selector

### LINKED ISSUE #25010: .matches(':defined') on an undefined custom element throws SyntaxError
WPT custom-elements/Document-createElement.html sees this, in `assert_false(instance2.matches(':defined'))`

### PR DESCRIPTION
**Pull request by pshaughn**
ElementState::IN_DEFINED_STATE already existed but wasn't hooked up to anything; now it is set appropriately, and CSS is aware of it. The main test for this selector needed the window named getter for no important reason, so I made a one-line test change to avoid that. Remaining failures in the test are all about SVG elements.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #25010 

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

File: components/style/servo/selector_parser.rs [hunk 1]
```diff
@@ -436,6 +439,7 @@ impl<'a, 'i> ::selectors::Parser<'i> for SelectorParser<'a> {
             "active" => Active,
             "any-link" => AnyLink,
             "checked" => Checked,
+        "defined" => Defined,
```
Comment (emilio) [inline components/style/servo/selector_parser.rs]: indentation here seems broken.

File: tests/wpt/web-platform-tests/custom-elements/pseudo-class-defined.html [hunk 2]
```diff
@@ -16,6 +16,7 @@
 const neither = 'rgb(255, 0, 0)';
 const defined = 'rgb(255, 165, 0)';
 const not_defined = 'rgb(0, 0, 255)';
+var iframe = document.getElementById("iframe"); // don't need to test window named getter here
```
Comment (emilio) [inline tests/wpt/web-platform-tests/custom-elements/pseudo-class-defined.html]: Maybe `const` for consistency with above?

File: components/style/servo/selector_parser.rs (hunk 1 above)
Comment (pshaughn) [reply components/style/servo/selector_parser.rs]: That's weird! I ran fmt and tidy and didn't look closely at the result; maybe it's a bug in fmt.

File: tests/wpt/web-platform-tests/custom-elements/pseudo-class-defined.html (hunk 2 above)
Comment (pshaughn) [reply tests/wpt/web-platform-tests/custom-elements/pseudo-class-defined.html]: You are correct; I'm just still in a pre-ES6 habit of blindly typing "var" for everything.

**Comment by jdm** [comment]
@bors-servo retry
* #22212

**Comment by pshaughn** [comment]
Oh, layout2020 also needs to know about the new enum value, I did not think of that.

**Comment by jdm** [comment]
This will need a rebase.

**Comment by jdm** [comment]
@bors-servo retry
* #24622

**Comment by CYBAI** [comment]
@bors-servo retry

- #25513 

```
  ▶ PASS [expected FAIL] /_mozilla/css/border-image-linear-gradient.html
```

**Comment by CYBAI** [comment]
@bors-servo retry

- #25513

**Pull request merged**